include/thread_pool/thread_pool.hpp

```diff
@@ -7,8 +7,11 @@
 #include <vector>
 #include <type_traits>
 #include <cassert>
+#include <future>
 #include "./worker.hpp"
 
+#define _tp_unused(x) ((void)(x))
```

Comment:
I'm not a fan of this macro. I think that `(void) something;` is idiomatic C++ and doesn't require a macro.